flow/layers/aiks_layer.cc

```diff
@@ -0,0 +1,64 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/flow/layers/aiks_layer.h"
+
+#include <utility>
+
+namespace flutter {
+
+AiksLayer::AiksLayer(const SkPoint& offset,
+                     const std::shared_ptr<const impeller::Picture>& picture)
+    : offset_(offset), picture_(picture) {
+#if IMPELLER_SUPPORTS_RENDERING
+  if (picture_) {
+    FML_DCHECK(picture_->rtree);
+    bounds_ = picture_->rtree->bounds();
+  }
+#endif
+}
+
+bool AiksLayer::IsReplacing(DiffContext* context, const Layer* layer) const {
+  auto old_layer = layer->as_aiks_layer();
+  return old_layer != nullptr && offset_ == old_layer->offset_ &&
+         old_layer->picture_ == picture_;
+}
+
+void AiksLayer::Diff(DiffContext* context, const Layer* old_layer) {
+  DiffContext::AutoSubtreeRestore subtree(context);
+  context->PushTransform(SkMatrix::Translate(offset_.x(), offset_.y()));
+  context->AddLayerBounds(bounds_);
+  context->SetLayerPaintRegion(this, context->CurrentSubtreeRegion());
+}
+
+void AiksLayer::Preroll(PrerollContext* frame) {
+  // There is no opacity peepholing to do here because Impeller handles that
+  // in the Entities, and this layer will never participate in raster caching.
+  FML_DCHECK(!frame->raster_cache);
+  set_paint_bounds(bounds_);
+}
+
+void AiksLayer::Paint(PaintContext& context) const {
+  FML_DCHECK(needs_painting(context));
+
+  auto mutator = context.state_stack.save();
+  mutator.translate(offset_.x(), offset_.y());
+
+  FML_DCHECK(!context.raster_cache);
+
+  SkScalar opacity = context.state_stack.outstanding_opacity();
```

Comment:
You never set the flags to indicate that you could render your contents with opacity, so the state stack will do a protective saveLayer around you instead and this value will always be 1.0.

If AiksCanvas.DrawImpellerPicture can distribute opacity, then you will need to set the appropriate flag in Preroll to be able to do this.

Reply:
It can't

Reply:
(but we could certainly update that later, or see about using the new split you created to handle that right?)